python/pyarrow/pandas_compat.py

```diff
@@ -194,73 +195,76 @@ def construct_metadata(df, column_names, index_levels, index_column_names,
     ----------
     df : pandas.DataFrame
     index_levels : List[pd.Index]
-    presere_index : bool
+    index_descriptors : List[Dict]
+    preserve_index : bool
     types : List[pyarrow.DataType]
 
     Returns
     -------
     dict
     """
+    num_serialized_index_levels = len([descr for descr in index_descriptors
+                                       if descr['kind'] == 'serialized'])
```

Comment:
`'kind'` is a new attribute that is not set in old metadata descriptions. Is this somewhere handled?

Reply:
I need to make a PR to resolve the pandas issue adding this. I have implemented backwards compatibility already for this, but I will add some unit tests to assert